cmd/auth-rpc-client.go

```diff
@@ -233,18 +247,17 @@ const connectSuccessMessage = "200 Connected to Go RPC"
 
 // dial tries to establish a connection to serverAddr in a safe manner.
 // If there is a valid rpc.Cliemt, it returns that else creates a new one.
-func rpcDial(serverAddr, serviceEndpoint string, secureConn bool) (netRPCClient *rpc.Client, err error) {
+func rpcDial(serverAddr, serviceEndpoint string, secureConn bool) (netRPCClient *rpc.Client, conn net.Conn, err error) {
```

Comment:
Returning `rpc.Client` and `net.Conn` look unclean.  You could return `net.Conn` and encapsulate  `rpc.Client` within `authClient`

Reply:
Sure will do that.